src/pages/Location/UserFieldDialog/UserFieldDialog.js

```diff
@@ -0,0 +1,70 @@
+import {
+  Dialog,
+  Button,
+  DialogTitle,
+  DialogContent,
+  DialogActions,
+  TextField,
+} from '@mui/material';
+import React from 'react';
+
+const UserFieldDialog = ({ fieldDialogState, setFieldDialogState, handleClose }) => {
+  const {
+    open, fieldName, error, errorText, actionType,
+  } = fieldDialogState;
+  return (
+    <Dialog sx={{ m: 0, p: 2 }} open={open}>
+      {actionType === 'add' && (
+      <>
+        <DialogTitle>
+          You need to give this field a nickname
+        </DialogTitle>
+        <DialogContent>
+          <TextField
+            autoFocus
+            error={error}
+            helperText={errorText}
+            variant="standard"
+            label="Field Nickname"
+            value={fieldName}
+            onChange={(e) => setFieldDialogState({ ...fieldDialogState, fieldName: e.target.value })}
+          />
+        </DialogContent>
+      </>
+      )}
+      {actionType === 'update' && (
```

Comment:
I think we just need to use the actionType to set a const string and then just bring that string into the DialogTitle and create a conditional just to display the TextField or not.  this will reduce a few lines and conditionals.